src/ui/utils.ts

```diff
@@ -1,7 +1,134 @@
-import {isMobile} from '../utils/platform';
+import {isMobile, isFirefox} from '../utils/platform';
+import UserStorage from '../background/user-storage';
 
 declare const __CHROMIUM_MV3__: boolean;
 
+declare const browser: {
+    theme: {
+        getCurrent: (() => Promise<void>);
+    };
+};
+
+export async function getTheme() {
+    // Check if theme setting should be used
+    await UserStorage.loadSettings();
+    if (UserStorage.settings.themeWithFirefox && isFirefox && typeof browser !== 'undefined' && browser.theme && browser.theme.getCurrent) {
+        setStyle(await browser.theme.getCurrent());
+    }
+}
+
+function parseColor(color: any) {
+    // Create a div, set a color, get it back in unified rgb format
+    var div = document.createElement('div'), m;
+    document.body.appendChild(div);
+    div.style.color = color;
+    m = getComputedStyle(div).color.match(/^rgb\s*\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)$/i);
+    div.remove();
+    if(m) return [Number(m[1]),Number(m[2]),Number(m[3])];
+    return [0,0,0];
+}
+
+function setStyle(Theme: any) {
+    if (Theme.colors) {
+        // Get color or use fallback white
+        var Color = Theme.colors.popup_text;
```

Comment:
Using `const` or `let` is preferred over `var`. In JavaScript naming convention Capitalized variable names are used for classes and Enums.